AmazonFreeRTOS/Services/MqttProxy/Publish.swift

```diff
@@ -10,49 +20,33 @@ public struct Publish: Encborable, Decborable {
     ///     - payload: Mqtt payload.
     /// - Returns: A new SaveNetworkReq.
     public init(topic: String, msgID: Int, qoS: Int, payload: Data) {
+        messageType = MqttMessageType.publish.rawValue
         self.topic = topic
         self.msgID = msgID
         self.qoS = qoS
         self.payload = payload
     }
 
-    init?(dictionary: NSDictionary) {
-        guard let topic = dictionary.object(forKey: CborKey.topic.rawValue) as? String else {
-            return nil
-        }
-        self.topic = topic
+    private enum CodingKeys: String, CodingKey {
+        case messageType = "w" /// CborKey.type.rawValue
+        case topic = "u" /// CborKey.topic.rawValue
+        case msgID = "i" /// CborKey.msgID.rawValue
+        case qoS = "n" /// CborKey.qoS.rawValue
+        case payload = "k" /// CborKey.payload.rawValue
+    }
+
+    public init(from decoder: Decoder) throws {
+        let values = try decoder.container(keyedBy: CodingKeys.self)
```

Comment:
Do we need a `try` if the function `throws` ?

Reply:
try should be ok. It can still throws the error to the upper level. Only "try?" will eliminate the error.